opm/autodiff/ParallelRestrictedAdditiveSchwarz.hpp

```diff
@@ -0,0 +1,296 @@
+/*
+  Copyright 2015 Dr. Blatt - HPC-Simulation-Software & Services
+  Copyright 2015 Statoil AS
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#ifndef OPM_PARALLELRESTRICTEDADDITIVESCHWARZ_HEADER_INCLUDED
+#define OPM_PARALLELRESTRICTEDADDITIVESCHWARZ_HEADER_INCLUDED
+
+#include <dune/istl/preconditioner.hh>
+#include <dune/istl/paamg/smoother.hh>
+namespace Opm
```

Comment:
The lack of whitespace and indentation makes parts of this hard to read (which namespace am I in now).